controllers/tests/controller_server/host_definer/csi_node_watcher_test.py

```diff
@@ -0,0 +1,322 @@
+        self.assertIn(messages.NEW_KUBERNETES_NODE.format(self.fake_csi_node_info.name), self._mock_logger.records)
+
+    def test_add_node_not_update_labels(self):
+        self.csi_node_watcher.csi_nodes_api.get.return_value = self.fake_k8s_csi_nodes_with_ibm_block
+        self.mock_os.getenv.return_value = settings.TRUE_STRING
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_manage_node_label
+        self.csi_node_watcher.add_initial_csi_nodes()
+        self.csi_node_watcher.core_api.patch_node.assert_not_called()
+        self.assertEqual(1, len(self.mock_nodes_on_watcher_helper))
+
+    def test_add_node_update_labels(self):
+        self.csi_node_watcher.csi_nodes_api.get.return_value = self.fake_k8s_csi_nodes_with_ibm_block
+        self.mock_os.getenv.return_value = settings.TRUE_STRING
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_fake_label
+        self.csi_node_watcher.add_initial_csi_nodes()
+        self.csi_node_watcher.core_api.patch_node.assert_called()
+        self.assertEqual(1, len(self.mock_nodes_on_watcher_helper))
+
+    def test_add_node_update_labels_log_message(self):
+        self.csi_node_watcher.csi_nodes_api.get.return_value = self.fake_k8s_csi_nodes_with_ibm_block
+        self.mock_os.getenv.return_value = settings.TRUE_STRING
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_fake_label
+        self.csi_node_watcher.add_initial_csi_nodes()
+        self.assertIn(messages.ADD_LABEL_TO_NODE.format(settings.MANAGE_NODE_LABEL,
+                      settings.FAKE_NODE_NAME), self._mock_logger.records)
+
+    def test_failed_update_node_label(self):
+        self.csi_node_watcher.csi_nodes_api.get.return_value = self.fake_k8s_csi_nodes_with_ibm_block
+        self.mock_os.getenv.return_value = settings.TRUE_STRING
+        self.csi_node_watcher.core_api.read_node.return_value = self.fake_k8s_node_with_fake_label
+        self.csi_node_watcher.core_api.patch_node.side_effect = self.fake_api_exception
+        self.csi_node_watcher.add_initial_csi_nodes()
+        utils.assert_fail_to_update_label_log_message(self.http_resp.data, self._mock_logger.records)
+        self.assertEqual(1, len(self.mock_nodes_on_watcher_helper))
+
+    def test_failed_to_get_csi_nodes(self):
+        self.csi_node_watcher.csi_nodes_api.get.side_effect = self.fake_api_exception
+        self.csi_node_watcher.add_initial_csi_nodes()
+        self.assertIn(messages.FAILED_TO_GET_CSI_NODES.format(self.http_resp.data), self._mock_logger.records)
+
+
+class TestWatchCsiNodesResources(BaseSetUp):
+    def setUp(self):
+        super().setUp()
+        self.csi_node_watcher._get_k8s_object_resource_version = Mock()
+        self.csi_node_watcher._get_k8s_object_resource_version.return_value = settings.FAKE_RESOURCE_VERSION
+        self.csi_node_watcher._delete_host_definition = Mock()
```

Comment:
please mock the third party APIs and not the class functions

Reply:
done